src/main/java/emissary/util/io/FileFind.java

```diff
@@ -91,7 +92,7 @@ class FileIterator implements Iterator {
         /**
          * Stack of Files and directory lists keeping track of where in the tree we are.
          */
-        private Stack<Object> currentPath = new Stack<>();
+        private List<Object> currentPath = new ArrayList<>();
```

Comment:
```suggestion
        private Deque<Object> currentPath = new ArrayDeque <>();
```

then you can revert other changes in this file